api/lib/block-listeners/cosmos-node-subscription.js

```diff
@@ -149,14 +171,6 @@ class CosmosNodeSubscription {
         await cosmosAPI.newBlockHandler(block, this.store)
       }
 
-      const validators = await cosmosAPI.getAllValidators(block.height)
```

Comment:
this would stay here

Reply:
why? Isn't it better to have it in a function on its own? Like we have proposals. There is no difference

Reply:
ah, I get it, because maybe in that block the validator got slashed or something like that no? ok :+1: putting it back

Reply:
Ah, but the problem is that I don't want to store validators without popularity. So yeah, I will have to do as you said and add popularity inside the store

Reply:
Moving it to the store made it terribly slow and created connection issues :shrug:   The fastest I have seen is with my last commit "applu one single DB connection". 

In the store the `update` function becomes async and everything slows down by x100

Reply:
In the store what changes mainly is this, and this really kills performance it seems:

```
  async update({
    height,
    block = this.block,
    validators = this.validators,
    data = this.data // multi purpose block to be used for any chain specific data
  }) {
    if (Array.isArray(validators)) {
      // add popularity field
      validators = await this.addPopularityToValidators(
        validators,
        this.network.id
      )
      // convert to map
      validators = this.getValidatorMap(validators)
    } else {
      console.error(`Validators should be passed to store as an Array`)
    }
```

Reply:
Was this before or after you fixed the 200 requests to the db issue?

Reply:
after

Reply:
I mean, I can push my tentative commit so you can see it with your own eyes, although I will probably need to later remove it

Reply:
I would like to cross check, if that is ok for you. Can you push?

Reply:
I will open another branch with it :wink:

Reply:
Here it is #4294